site/en/docs/extensions/mv3/getstarted/development-basics/index.md

```diff
@@ -0,0 +1,269 @@
+
+{% Label %}hello.html{% endLabel %}
+
+```html
+<html>
+  <body>
+    <h1>Hello Extensions of the world! 👋</h1>
+    <script src="popup.js"></script>
+  </body>
+</html>
+```
+
+Create a `popup.js` file and add the following code:
+
+{% Label %}popup.js{% endLabel %}
+
+```js
+console.log("This is a popup!")
+```
+
+To see this message logged in the Console:
+
+  1. Refresh the extension.
+  2. Open the popup.
+  3. Right-click on the popup.
+  4. Select **Inspect**. 
+      <figure> 
+      {% Img src="image/BhuKGJaIeLNPW9ehns59NfwqKxF2/vHGHW1o4J0kZgUkAteRQ.png", 
+      alt="Inspecting the popup", width="322", height="262", class="screenshot" %}
+        <figcaption>
+        Inspecting a popup 
+        </figcaption>
+      </figure>
+  5. In the [DevTools][dev-tools], navigate to the **Console** panel.
+    <figure>
+    {% Img src="image/BhuKGJaIeLNPW9ehns59NfwqKxF2/1ZrcBEYcbMxW1c9UvBy9.png", 
+    alt="DevTools Code Panel", width="400", height="374", class="screenshot" %}
+      <figcaption>
+      DevTools Console Panel 
+      </figcaption>
+    </figure>
+
+### Error logs {: #errors }
+
+Now let's break the extension. We can do so by removing the closing quote in the `popup.js` file:
+
+{% Label %}popup.js{% endLabel %}
+
+```js
+console.log("This is a popup!) // broken code
+```
+
+Go to the Extension Management page and open the popup. An **Errors** button will appear.
+
+{% Img src="image/BhuKGJaIeLNPW9ehns59NfwqKxF2/a9lAHCJZZrebOSKrkPRD.png", 
+alt="Extension Management page with error button", width="300", height="226", class="screenshot" %}
+
+Click the **Errors** button to learn more about the error:
+
+{% Img src="image/BhuKGJaIeLNPW9ehns59NfwqKxF2/GrorLyaC6nRAyqc1qULC.png", 
+alt="Extension error details", width="400", height="281", class="screenshot" %}
+
+To learn more about debugging the service worker, options page, and content scripts, see [Debugging
+extensions][doc-debug].
+
+## Structuring an extension project {: #structure }
+
+There are many ways to structure an extension project; however, you must place the
+manifest.json file in the extension's **root directory**. The following is a structure example:
+
+
+```text/1
+└── my-extension/
+    ├── manifest.json
+    ├── background.js
+    ├── scripts/
+    │   └── content.js
+    │   └── react.production.min.js
+    ├── popup/
+    │   └── popup.html
+    │   └── popup.js
+    │   └── popup.css
+    └── images/
+        ├── icon-16.png
+        ├── icon-32.png
+        ├── icon-48.png
+        └── icon-128.png
+```
+
+## Using Typescript {: #types }
+
+If you are developing using a [code editor][mdn-ide] such as VSCode or Atom, you can use the npm
+package [chrome-types][npm-chrome-types] to take advantage of auto-completion for the [Chrome
+API][doc-apis]. This npm package is updated automatically when the Chromium source code
+changes.
+
+{% Aside  'gotchas' %}
+
+Upgrade this npm package frequently to work with the latest Chromium version.
+
+{% endAside %}
```

Comment:
> {% Aside  'gotchas' %}
> 
> Upgrade this npm package frequently to work with the latest Chromium version.
> 
> {% endAside %}
> 

@jpmedley I had to change this to gotchas, because we don't have important in this repo yet. See [Issue#3380](https://github.com/GoogleChrome/developer.chrome.com/issues/3380)